### PR TITLE
Porting shadow manager

### PR DESCRIPTION
**Pull request by jiangpta**
**Issue #, if available:**
  n/a

**Description of changes:**
  Add a test case with a simple scenario of resetting default fields before deployment.

**Why is this change necessary:**
  Add uat tests to shadowmanager

**How was this change tested:**
  Tested in remote EC2 linux instance with result:
      [Given my device is registered as a Thing....................................passed
      And my device is running Greengrass.........................................passed
      When I create a Greengrass deployment with components.......................passed
      And I update my Greengrass deployment configuration, setting the component aws.greengrass.ShadowManager configuration to:.passed
      And I update my Greengrass deployment configuration, setting the component aws.greengrass.Nucleus configuration to:.passed
      And I deploy the Greengrass deployment configuration........................passed
      Then the Greengrass deployment is COMPLETED on the device after 120 minutes.passed

**Any additional information or context required to review the change:**
   This test case currently contains only basic step, more steps and scenarios will be added in this test case.

**Checklist:**
 - [x] Updated the README if applicable
 - [ ] Updated or added new unit tests
 - [ ] Updated or added new integration tests
 - [ ] If your code makes a remote network call, it was tested with a proxy
 
By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by Nelsonochoam — DISMISSED** [review]

File: uat/src/main/resources/greengrass/features/shadow-6.feature [hunk 1]
```diff
@@ -0,0 +1,31 @@
+@ShadowManager @Shadow-6 @Greengate
```
Comment (saranyailla) [inline uat/src/main/resources/greengrass/features/shadow-6.feature:1]: Remove Greengate tag

File: uat/src/main/resources/greengrass/features/shadow-6.feature [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+            "maxDiskUtilizationMegaBytes":16
+         }
+      }
+      """
+    And I update my Greengrass deployment configuration, setting the component aws.greengrass.Nucleus configuration to:
+        """
+        {"MERGE":{"logging": {"level": "DEBUG"}}}
+        """
+    And I deploy the Greengrass deployment configuration
+    Then the Greengrass deployment is COMPLETED on the device after 2 minutes
```
Comment (saranyailla) [inline uat/src/main/resources/greengrass/features/shadow-6.feature:31]: The test is incomplete.  Please include all the steps and update the description as you see in EvergreenFeatures package.

File: uat/src/main/resources/greengrass/features/shadow-6.feature (hunk 1 above)
Comment (jiangpta) [reply uat/src/main/resources/greengrass/features/shadow-6.feature:1]: done

File: uat/src/main/resources/greengrass/features/shadow-6.feature (hunk 2 above)
Comment (jiangpta) [reply uat/src/main/resources/greengrass/features/shadow-6.feature:61]: done.

**Review by saranyailla — DISMISSED** [review]
We'll be adding local shadow operations to verify these steps in the later PRs.

File: uat/src/main/resources/greengrass/features/shadow-6.feature [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+
```
Comment (Nelsonochoam) [inline uat/src/main/resources/greengrass/features/shadow-6.feature:36]:
```suggestion
```

File: uat/src/main/resources/greengrass/features/shadow-6.feature [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+    When I update my Greengrass deployment configuration, setting the component aws.greengrass.ShadowManager configuration to:
+    """
+    {"MERGE":{"strategy":{"type":"realtime"}}}
+    """
+    And I deploy the Greengrass deployment configuration
+    Then the Greengrass deployment is COMPLETED on the device after 2 minutes
+
```
Comment (Nelsonochoam) [inline uat/src/main/resources/greengrass/features/shadow-6.feature:43]:
```suggestion
    Then I verify the aws.greengrass.ShadowManager component is RUNNING using the greengrass-cli
```

File: uat/src/main/resources/greengrass/features/shadow-6.feature [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+        "synchronize":{
+          "maxOutboundSyncUpdatesPerSecond":50
+        },
+        "shadowDocumentSizeLimitBytes":8192,
+        "maxDiskUtilizationMegaBytes":16
+      }
+    }
+    """
+    And I deploy the Greengrass deployment configuration
+    Then the Greengrass deployment is COMPLETED on the device after 2 minutes
```
Comment (Nelsonochoam) [inline uat/src/main/resources/greengrass/features/shadow-6.feature:61]:
```suggestion
    Then the Greengrass deployment is COMPLETED on the device after 2 minutes
    Then I verify the aws.greengrass.ShadowManager component is RUNNING using the greengrass-cli
```

File: uat/src/main/resources/greengrass/features/shadow-6.feature (hunk 3 above)
Comment (jiangpta) [reply uat/src/main/resources/greengrass/features/shadow-6.feature:36]: done

**Review by Nelsonochoam — APPROVED** [review]

**Review by saranyailla — APPROVED** [review]

**Pull request merged**